### PR TITLE
[autoscaler] Ray Up url-arg

### LINKED ISSUE #6993: Allow `ray up` to take URL as argument.
To get started with the autoscaler, we tell people to run a line like `ray up ray/python/ray/autoscaler/aws/example-full.yaml`. In practice, the steps are

1. Figure out that this means you have to clone the Ray repository.
2. Clone the Ray repository.
3. Figure out the new correct relative file path based on where you cloned it.
4. Run `ray up ...`.

Instead, we could let people just copy and paste the following command:

```
ray up https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml
```

In this blog post https://towardsdatascience.com/10x-faster-parallel-python-without-python-multiprocessing-e5017c93cce1, I posted benchmarks at the end that they could run with the autoscaler, but they still have to `wget` the relevant file. It'd be much easier if they could just use the URL.

Could also just write stuff like the following, but one line would be better.

```
wget https://raw.githubusercontent.com/ray-project/ray/master/python/ray/autoscaler/aws/example-full.yaml
ray up example-full.yaml
```

### PR DESCRIPTION
**Pull request by JosephTLucas**
Ray up will check the first argument to see if a local file exists, if not it will attempt to fetch the file as a url.  This seemed like the least intrusive to the current functionality and precluded the need for any additional arguments.  If it is not robust enough, I could include a url verification check before I attempt to fetch the file, but either way that's a failure for the user.
Unit Test added and passed in the autoscaling yaml test file, but requires a destination file, so I used one from your repository.  Not sure if a unit test requiring internet connectivity is a deal breaker.

This is my first attempt at contributing to open source.  So I appreciate any programming/contributing feedback.

## Why are these changes needed?
This PR is in response to https://github.com/ray-project/ray/issues/6993

## Related issue number
Closes #6993 

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.  I can do this if you like/accept my solution
- [X] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [X] Unit tests:  Added one
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by rkooo567** [comment]
Hi, @JosephTLucas Thanks for your contribution! It seems like this PR has linting error. Would you mind running `ci/travis/format.sh` and make sure every linting errors are resolved?

Thank you!

**Comment by JosephTLucas** [comment]
> Hi, @JosephTLucas Thanks for your contribution! It seems like this PR has linting error. Would you mind running `ci/travis/format.sh` and make sure every linting errors are resolved?

Working on this, but getting some errors (flake8: error: no such option: --inline-quotes).  I'll troubleshoot/research, but wanted to update on the reason for the delay.

**Comment by richardliaw** [comment]
`pip install flake8-quotes`?

**Comment by JosephTLucas** [comment]
Thanks @richardliaw !  I have never used flake before and had experimented with just installing flake8.  Didn't realize there was a specific quotes package.

I was able to run the linter and see one of my strings was too long, so I made that change.  Now I see `ray_env/lib/python3.6/site-packages/numpy/testing/_private/utils.py:2460 import psutil without explicitly import ray before it.` which I believe is unrelated to any code I wrote.  Is there a way to only run the linter against my diffs or is that error message actually indicative of a change I made?

As of now, I know I changed one of my lines that offended the linter, but can't tell if there were others.  Thanks for the patience.  Obviously new to this.

**Comment by richardliaw** [comment]
No problem! Questions are welcome. Try running `scripts/format.sh` from the Ray repository.

File: python/ray/tests/test_autoscaler_yaml.py [hunk 1]
```diff
@@ -25,6 +26,24 @@ def testValidateDefaultConfig(self):
             except Exception:
                 self.fail("Config did not pass validation test!")
 
+    def testValidateNetworkConfig(self):
+        web_yaml = "https://raw.githubusercontent.com/ray-project/ray/" \
+            "master/python/ray/autoscaler/aws/example-full.yaml"
+        config_path = os.path.join(RAY_PATH, "autoscaler",
+                                   "downloaded_cluster_config.yaml")
+        response = urllib.request.urlopen(web_yaml, timeout=5)
```
Comment (richardliaw) [inline python/ray/tests/test_autoscaler_yaml.py:34]: how about dumping a unique location in tmpdir? (unique to the URL)

File: python/ray/tests/test_autoscaler_yaml.py [hunk 2]
```diff
@@ -25,6 +26,24 @@ def testValidateDefaultConfig(self):
...
+        content = response.read()
+        f = open(config_path, "wb")
```
Comment (richardliaw) [inline python/ray/tests/test_autoscaler_yaml.py:36]: what about `with open` so we don't need to explicitly f.close?

File: python/ray/scripts/scripts.py [hunk 3]
```diff
@@ -562,6 +563,16 @@ def create_or_update(cluster_config_file, min_workers, max_workers, no_restart,
     if restart_only or no_restart:
         assert restart_only != no_restart, "Cannot set both 'restart_only' " \
             "and 'no_restart' at the same time!"
+    if not os.path.isfile(cluster_config_file):
+        try:
+            response = urllib.request.urlopen(cluster_config_file, timeout = 5)
+            content = response.read()
+            f = open("downloaded_cluster_config.yaml", "wb")
+            f.write(content)
+            f.close()
```
Comment (richardliaw) [inline python/ray/scripts/scripts.py:572]: maybe use `with open`?

And then for the file name, can you use the file name from the original file?

File: python/ray/scripts/scripts.py [hunk 4]
```diff
@@ -562,6 +563,16 @@ def create_or_update(cluster_config_file, min_workers, max_workers, no_restart,
     if restart_only or no_restart:
         assert restart_only != no_restart, "Cannot set both 'restart_only' " \
             "and 'no_restart' at the same time!"
+    if not os.path.isfile(cluster_config_file):
```
Comment (richardliaw) [inline python/ray/scripts/scripts.py:566]: is it possible to check if this is a path?

**Comment by JosephTLucas** [comment]
Great feedback, thanks! I’ll make those changes.

**Comment by richardliaw** [comment]
No problem, just @ me when you're done. Thanks a bunch for opening up this pr; really appreciate it!

**Comment by JosephTLucas** [comment]
@richardliaw Changes made (and learned a bit in the process!).  Ran it through the linter again (still getting a numpy import ordering error, but nothing from my stuff).

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
Can you rebase off master? Looks like all the tests are failing?

**Comment by JosephTLucas** [comment]
@richardliaw I conceptually understand the rebase, but not sure if I executed correctly.  I think I was able to successfully test with the commits made to Ray-master since my initial PR.

**Comment by richardliaw** [comment]
Looks good, man!

**Pull request merged**